Sources/Basics/HTPClient+URLSession.swift

```diff
@@ -8,28 +8,120 @@ import struct TSCUtility.Versioning
 import FoundationNetworking
 #endif
 
-public struct URLSessionHTTPClient: HTTPClientProtocol {
```

Comment:
should we extract these to separate PR, or you prefer to bake here more first?

Reply:
Yeah, these should be in a separate PR. I just needed them here and now to unblock myself.